### PR TITLE
Remotes/origin/add tailwind and tailwind UI

### PR DESCRIPTION
**Pull request by ShawkiS**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,18 +15,21 @@
...
   "scripts": {
     "dev:chrome": "cross-env NODE_ENV=development cross-env TARGET_BROWSER=chrome webpack --watch",
+    "build:style": "postcss src/styles/tailwind.css -o src/styles/index.css",
     "dev:firefox": "cross-env NODE_ENV=development cross-env TARGET_BROWSER=firefox webpack --watch",
     "dev:opera": "cross-env NODE_ENV=development cross-env TARGET_BROWSER=opera webpack --watch",
     "build:chrome": "cross-env NODE_ENV=production cross-env TARGET_BROWSER=chrome webpack",
     "build:firefox": "cross-env NODE_ENV=production cross-env TARGET_BROWSER=firefox webpack",
     "build:opera": "cross-env NODE_ENV=production cross-env TARGET_BROWSER=opera webpack",
-    "build": "yarn run build:chrome && yarn run build:firefox && yarn run build:opera",
+    "build": "craco build && npm run build:style && yarn run build:chrome && yarn run build:firefox && yarn run build:opera",
```
Comment (bumi) [inline package.json:24]: why do we need that?
can this be part of webpack? 
and if we need it we probably also need it in the other `build` tasks?

File: src/styles/tailwind.css [hunk 2]
```diff
@@ -0,0 +1,5 @@
+@tailwind base;
```
Comment (bumi) [inline src/styles/tailwind.css:1]: I think `styles` should go under `app/styles` as the `app` holds the user facing application

File: package.json [hunk 3]
```diff
@@ -84,6 +99,7 @@
     "copy-webpack-plugin": "^8.1.1",
     "cross-env": "^7.0.3",
     "css-loader": "^5.2.0",
+    "dotenv-webpack": "^7.0.2",
```
Comment (bumi) [inline package.json:102]: this is new, isn't it? where do we use env/.env configurations?

**Review by bumi — COMMENTED** [review]
nice! super cool. 
I really like storybook I have to say :)

File: package.json (hunk 1 above)
Comment (ShawkiS) [reply package.json:24]: I made a change for it. It generates the tailwind css file, we don't need it in every build, just once or when adding new things

File: src/styles/tailwind.css (hunk 2 above)
Comment (ShawkiS) [reply src/styles/tailwind.css:1]: right. just moved it.

File: package.json (hunk 3 above)
Comment (ShawkiS) [reply package.json:102]: hmmmmm yeah it seems came by a mistake

File: package.json (hunk 1 above)
Comment (bumi) [reply package.json:24]: ah ok, thanks.

and the `craco build` ? is this also only needed for the styles? (if so, can we add it to the `build:style` command?)

File: package.json [hunk 4]
```diff
@@ -15,18 +15,21 @@
...
   "scripts": {
     "dev:chrome": "cross-env NODE_ENV=development cross-env TARGET_BROWSER=chrome webpack --watch",
+    "build:style": "tailwind build src/app/styles/tailwind.css -o src/app/styles/index.css",
     "dev:firefox": "cross-env NODE_ENV=development cross-env TARGET_BROWSER=firefox webpack --watch",
     "dev:opera": "cross-env NODE_ENV=development cross-env TARGET_BROWSER=opera webpack --watch",
     "build:chrome": "cross-env NODE_ENV=production cross-env TARGET_BROWSER=chrome webpack",
     "build:firefox": "cross-env NODE_ENV=production cross-env TARGET_BROWSER=firefox webpack",
     "build:opera": "cross-env NODE_ENV=production cross-env TARGET_BROWSER=opera webpack",
-    "build": "yarn run build:chrome && yarn run build:firefox && yarn run build:opera",
+    "build": "craco build && npm run build:style && yarn run build:chrome && yarn run build:firefox && yarn run build:opera",
```
Comment (bumi) [inline package.json:24]:
```suggestion
    "build": "yarn run build:style && yarn run build:chrome && yarn run build:firefox && yarn run build:opera",
```

File: craco.config.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (bumi) [inline craco.config.js:1]: this file is not needed, correct?

Comment (ShawkiS) [reply craco.config.js:1]: yes

File: package.json [hunk 6]
```diff
@@ -41,23 +44,30 @@
   "dependencies": {
     "@ant-design/icons": "^4.6.2",
     "@babel/runtime": "^7.13.10",
+    "@craco/craco": "^6.1.2",
```
Comment (bumi) [inline package.json:47]: I missed this before. Are we using craco? 

can you review that only required dependencies are added?

**Comment by bumi** [comment]
thanks for the PR! 

btw. we use yarn. so we should not commit the package-lock.json file afaik .

**Pull request merged**